src/plugins/discourse/config.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import {type MirrorOptions} from "./mirror";
+
+export type DiscourseConfig = {|
+  +serverUrl: string,
+  +mirrorOptions?: $Shape<MirrorOptions>,
+|};
+
+type JsonObject =
+  | string
+  | number
+  | boolean
+  | null
+  | JsonObject[]
+  | {[string]: JsonObject};
+
+export function parseConfig(raw: JsonObject): DiscourseConfig {
+  if (raw == null || typeof raw !== "object" || Array.isArray(raw)) {
+    throw new Error("bad config: " + JSON.stringify(raw));
+  }
+  let mirrorOptions = undefined;
+  const {serverUrl} = raw;
+  if (typeof serverUrl !== "string") {
+    throw new Error("serverUrl not string: " + JSON.stringify(serverUrl));
+  }
+  const httpRE = new RegExp(/^https?:\/\//);
+  if (!httpRE.test(serverUrl)) {
+    throw new Error(
+      "expected server url to start with 'https://' or 'http://'"
+    );
+  }
+  if (raw.mirrorOptions !== undefined) {
```

Comment:
As noted, this code lets the `mirrorOptions` be omitted, but it doesn’t
let it be partial: if the object exists, so must all its keys. This is
sound, but it’s more strict than the `$Shape<MirrorOptions>` would
require. Okay with me to use that strict behavior for now, especially
because it’ll get cleaned up shortly.